model/src/main/resources/defaults.yml

```diff
@@ -0,0 +1,358 @@
+# Copyright (C) 2022 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+# License-Filename: LICENSE
+
+# A file with default configuration options used by ORT. These options can be overridden in custom configuration files.
+
+ort:
+  variablesToPropagate:
```

Comment:
Phew. Looking at the length of this, I really wonder whether the "allow"-approach is the right one.

In either case we should probably consider to support wildcards or regexes, so one could allow / deny e.g. `CARGO_*`.

Reply:
Yes, I was already sceptical when the decision was taken. It will be very difficult to come up with an exhaustive list. In addition, this list will have to be maintained when there are new versions of tools.

Using of wildcards will simplify the configuration, but again increase the risk that an unwanted variable is passed through.

Reply:
I wonder whether we should come up with something completely different... a "crazy" approach could be to look at the entropy of the value of an environment variable to "guess" whether it's likely a password or so, and only then do not pass it on to the spawned process.

Or, instead of a global list, create tool-specific lists that are used in `CommandLineTool` `object`s. I.e. we should wrap each tool we call in an `object` instead of implementing the interface in the main class itself (see [`MercurialCommand`](https://github.com/oss-review-toolkit/ort/blob/4d90096fb049344092a4c067edcc74f5d1472b58/downloader/src/main/kotlin/vcs/Mercurial.kt#L39-L52) for an example), and each of these objects gets its own (hard-coded) list of allowed environment variables. That should at least ease maintainability a bit.

Reply:
I fear the entropy approach is too clever to work in practice.

If the allowed variables were configured at the command line tools, this could indeed make the maintenance a bit easier. But how could these configurations then be extended? For our use case, we will need to pass additional variables with credentials to package managers.